plugins/client/grpc/client.go

```diff
@@ -100,8 +102,15 @@ func (c *Client) Prepare() error {
 		"client_name": Name,
 	})})
 
+	// server address resolver
+	resolvers.RegisterAllGrpcResolvers()
+
 	// connect to server
-	client, err := grpc.Dial(c.ServerAddr, *cfg...)
+	target, err := resolvers.BuildTarget(&c.ServerFinderConfig)
+	if err != nil {
+		return fmt.Errorf("cannot build grpc target: %v", err)
+	}
+	client, err := grpc.Dial(target, *cfg...)
```

Comment:
This makes me feel you have one logic client reference? Who controls the routine rule?